### ISSUE TITLE
Beta 20170221: Autoupload of video corrupts files

### ISSUE THREAD

**Issue by OlivierS1** [nextcloud/android#673]
### Actual behaviour
-Tell us what happens
When Autoupload is enabled for Camera folder, and option is set to Original files will be... "Moved to app folder" will cause any new video file being recorded to become corrupted.

### Expected behaviour
-Tell us what should happen
Nextcloud app should detect when the file has been finalized before moving ahead with the autoupload.

### Steps to reproduce
1. Have Autoupload for Camera to be enabled.
2. Set Original files will be... "Moved to app folder"
3. Open Camera app and take a video. 10-20seconds should be enough.
4. Transfer is probably underway while video is in progress (I suspect start of video file has been moved) but recording is still in progress.
5. File is successfully sent to server, or so it seems, but file is only 30-70MB in size and trying to play it returns error from player.

### Environment data
Android version:
6.0

Device model: 
LG G3

Stock or customized system:
Stock

Nextcloud app version:
Beta 201702221

Nextcloud server version:
NC 11.01

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```

**Comment by mario** [comment]
Hello,

unfortunately, it looks like all options apart from "Leave in original folder" are broken. I'll try to take a look, but just take note that the app can never really know when a file has finished being written to (there is inotify CLOSE_WRITE which is not reliable and doesn't work on every phone).

**Comment by mario** [comment]
Can you try with the:

https://github.com/nextcloud/android/blob/auto-upload-apk/apks/android-debug.apk?raw=true

Please? :)

**Comment by mario** [comment]
Actually the new APK won't fix it. Let me think about the video use case.

**Comment by mario** [comment]
Please test the new APK - just uploaded a new one :)

**Comment by OlivierS1** [comment]
I cannot install the debug apk for some reason.

**Comment by mario** [comment]
o.o What does it say? Remove the original client, and try this?

**Comment by OlivierS1** [comment]
I uninstalled my normal 1.4.1 Nextcloud app to be able to install the debug app. New version is set to 1.4.2.RC1.
It seems with this version that the upload is successful, however I have some comments:

1:  Under Upload menu, file size indicated is 110.3MB, but actual file size on the server for the same file is 254.4MB.
2: Under Autoupload options, I have "Remote folder" set to "/InstanUpload/Camera" and "Use subfolders" checked. However, the file will be saved with the following folder structure:
"/InstanUpload/Camera2017/02"
The year is not created as a single folder but is rather combined with Camera name.

**Comment by mario** [comment]
a) How important is that? I'm not sure I can fix this atm - so as long as the file works ... Can you check how big the file is on the filesystem?
b) Let me check...

**Comment by mario** [comment]
Fixed b) but won't build an APK now. Just answer me about a) :)

@enoch85 why did you set a milestone? :D

**Comment by OlivierS1** [comment]
If you mean a) corresponding to comment 1, then it is not crucially important as long as the file is reliably uploaded which was not the case before. But to be consistent and really meaningfull it would be good to have closer to reality numbers. I don't think those numbers are available before and during the upload process so at this time it is irrelevant.
I did however pay attention to the size after the fact which helped me realize what the issue was when failure took place.

**Comment by enoch85** [comment]
@mario This is a bug in the latest beta which should be fixed before 1.4.2 is released IMHO.

**Comment by mario** [comment]
@enoch85 maybe, but we need to agree on that - nobody assigns milestones on their own :)

Anyway ... the important bit (folder name) was fixed, and size shouldn't matter (it's really only a *very* minor UI bug). Is that ok? :)

**Comment by enoch85** [comment]
@mario You got a point. 

What's left for 1.4.2 then?

**Comment by mario** [comment]
Auto upload should be working quite well, so once that is approved + my filtering changes/fixes (should be soon), we're ready for a real RC.
